Code/Framework/AzFramework/Platform/Common/WinAPI/AzFramework/IO/LocalFileIO_WinAPI.cpp

```diff
@@ -33,7 +36,7 @@ namespace AZ
             char resolvedPath[AZ_MAX_PATH_LEN];
             ResolvePath(filePath, resolvedPath, AZ_MAX_PATH_LEN);
 
-            AZ::OSString searchPattern;
+            AZStd::string searchPattern;
```

Comment:
I was wondering the same, I changed it because below I need to use `wstring` and it has the same constraints
If I catch a case where SystemAllocator is not ready I need to fix it, so far no issues